app/presenters/schools/courses/certificates_presenter.rb

```diff
@@ -26,6 +27,11 @@ def self.certificate_details(certificate)
 
       private
 
+      def can_be_auto_issued
+        top_level = @course.levels.order(number: :desc).first.id
+        @course.target_groups.where(level_id:top_level,milestone:true).exists?
+      end
```

Comment:
@a-c-sreedhar-reddy: Two things to consider here: 
- You need not load the `id` of the highest level. Can scope target groups using the `highest_level` itself. `highest_level.target_groups.where...`
- There is a possibility that target groups can be archived, which is determined by the `archived` column. We should ensure that there is a live milestone target group in the highest level.

Reply:
Actually, since we're dealing with a single record, and not a relation, the most performant option would be to get the `id`. However, the `pick` method would work well here:

```ruby
highest_level_id = @course.levels.order(number: :desc).pick(:id)
@course.target_groups.where(level_id: highest_level_id, milestone: true).exists?
```